test/batch-transaction.ts

```diff
@@ -315,6 +347,7 @@ describe('BatchTransaction', () => {
         table: QUERY.table,
         keySet: fakeKeySet,
         dataBoostEnabled: true,
+        directedReadOptions: fakeDirectedReadOptionsForRequest,
       };
 
       const stub = sandbox.stub(batchTransaction, 'createPartitions_');
```

Comment:
Can we write a unit test to validate `transaction.execute(partition) `?
Basically check, if the ReadPartition or QueryPartition has the directed read options set, then the transaction.execute(partition), should make correct ReadRequest/ExecuteSqlRequest with directedReadOptions set in its request.

Reply:
In unit test we can not have e2e test. But I have added the test case for below scenarios , create partitions should return directedReadOptions and execute should take directedReadOptions